extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -184,7 +184,7 @@ export class Bind {
      * The current values of all bound expressions on the page.
      * @const @private {!JsonObject}
      */
-    this.state_ = /** @type {!JsonObject} */ (map());
+    this.state_ = dict();
```

Comment:
This was intentionally prototype-less. Alternatively we can try changing the implementation of `dict()` to return `opt_initial || map()`.

Reply:
i like this. this is probably even more preferable.

Reply:
cc @jridgewell if he thinks theres anything wrong with this change. i kinda prefer this since we "json" like objects shouldn't be manipulated like object anyways (using prototype methods etc)

Reply:
thinking about it some, there is an issue in changing it because the caller can set the initial object as well.

Reply:
To apply it to the initial object too: `opt_initial ? Object.assign(map(), opt_initial) || map()`.

Prototype functions like `toString` and `hasOwnProperty` won't work anymore, though we shouldn't be calling those on JSON anyways.

Reply:
Fine with me.